ecs-cli/modules/cli/compose/entity/service/service.go

```diff
@@ -523,13 +523,17 @@ func (s *Service) buildCreateServiceInput(serviceName, taskDefName string, desir
 	if s.healthCheckGP != nil && s.loadBalancers == nil {
 		return nil, fmt.Errorf("--%v is only valid for services configured to use load balancers", flags.HealthCheckGracePeriodFlag)
 	}
+
 	// TODO: revert to "LATEST" when latest refers to 1.4.0
-	platformVersion := aws.String("LATEST")
-	if len(ecsParams.TaskDefinition.EFSVolumes) != 0 {
-		log.Warnf("Detected an EFS Volume in task definition %s", taskDefName)
-		log.Warn("Using Fargate platform version 1.4.0, which includes changes to the networking flows for VPC endpoint customers.")
-		log.Warn("Learn more: https://aws.amazon.com/blogs/containers/aws-fargate-launches-platform-version-1-4/")
-		platformVersion = aws.String("1.4.0")
+	var platformVersion *string
+	if launchType == "FARGATE" {
```

Comment:
can we add a [unit test](https://github.com/aws/amazon-ecs-cli/blob/master/ecs-cli/modules/cli/compose/entity/service/service_test.go) (perhaps add assertions on existing EC2 service tests) which ensure that platform version is not set when launchType != FARGATE?

Reply:
I think it might be nice to have a separate unit test, like TestEFSWithEC2. We can do this by setting the launchtype in the CommandConfig within the createServiceTest helper: https://github.com/aws/amazon-ecs-cli/blob/d27908156e0f26d8851e1689cd2001be98833f21/ecs-cli/modules/cli/compose/entity/service/service_test.go#L1036

Reply:
Eg:

```
func TestCreateWithEFS_EC2(t *testing.T) {
	flagSet := flag.NewFlagSet("ecs-cli-up", 0)

	createServiceTest(
		t,
		flagSet,
		&config.CommandConfig{LaunchType: config.LaunchTypeEC2},
		ecsParamsWithFargateEFSVolume(),
		func(input *ecs.CreateServiceInput) {
			launchType := input.LaunchType
			assert.Equal(t, config.LaunchTypeFargate, aws.StringValue(launchType), "launch type is not fargate")
			platformVersion := input.PlatformVersion

			assert.Nil(t, platformVersion)
		},
		ecsSettingDisabled,
	)
}
```